library/waku_thread/inter_thread_communication/requests/node_lifecycle_request.nim

```diff
@@ -29,18 +28,22 @@ type
     STOP_NODE
 
 type
-  NodeLifecycleRequest* = ref object of InterThreadRequest
+  NodeLifecycleRequest* = object
     operation: NodeLifecycleMsgType
     configJson: cstring ## Only used in 'CREATE_NODE' operation
 
 proc new*(T: type NodeLifecycleRequest,
           op: NodeLifecycleMsgType,
-          configJson: cstring = ""): T =
+          configJson: cstring = ""): ptr NodeLifecycleRequest =
```

Comment:
```
proc create*(T: type Nodeli...): ptr T = ...
```

Reply:
actually, `createShared` because it's taken from the shared allocator

Reply:
I changed that, thanks! ( as a return type I set `ptr type T` )